piqe_ocp_lib/tests/resources/test_ocp_pods.py

```diff
@@ -0,0 +1,39 @@
+import logging
+
+import pytest
+
+from piqe_ocp_lib import __loggername__
+from piqe_ocp_lib.api.resources import OcpNodes, OcpPods
+
+logger = logging.getLogger(__loggername__)
+
+
+@pytest.fixture(scope="class")
+def setup_params(get_kubeconfig):
+    params_dict = {}
+    params_dict["pods_api_obj"] = OcpPods(kube_config_file=get_kubeconfig)
+    params_dict["nodes_api_obj"] = OcpNodes(kube_config_file=get_kubeconfig)
+    return params_dict
+
+
+class TestOcpPods:
+    def test_list_of_pods_in_a_node(self, setup_params):
+        """
+        Verify that list of pods are returned for a node_name
+        1. Call list_of_pods_in_a_node method via a ocp_pods instance
+        2. Verify len of list is not zero.
+        :param setup_params:
+        :return:
+        """
+        pods_api_obj = setup_params["pods_api_obj"]
+        nodes_api_obj = setup_params["nodes_api_obj"]
+        node_name_list = []
+        pod_exists = False
+        api_response = nodes_api_obj.get_worker_nodes()
+        for item in api_response.items:
+            node_name_list.append(item["metadata"]["name"])
+        for node in node_name_list:
+            pods_list = pods_api_obj.list_of_pods_in_a_node(node)
+            if len(pods_list) > 0:
```

Comment:
If I am reading this logic correctly, we check each worker node and make sure the list of pods on each one is greater than zero. If we have three worker nodes and the first two have pods but the last one doesn't, it looks like this test would fail, correct? I would enhance this test by storing the counts of pod_list length in a separate list and then perform an any() to make sure that only require at least one of the worker nodes to have pods on it. I'd be interested in hearing your thoughts.

Reply:
Correct, so it would check all 3 nodes and set the flag to true, if **any** one worker node returns pods and pods_list is non zero.

Reply:
I see, didn't read it correctly the first time. LGTM